File: src/lib/scala/Slate.Common/src/main/scala/slate/common/encrypt/Encryptor.scala
/**
<slate_header>
  author: Kishore Reddy
  url: https://github.com/kishorereddy/scala-slate
  copyright: 2015 Kishore Reddy
  license: https://github.com/kishorereddy/scala-slate/blob/master/LICENSE.md
  desc: a scala micro-framework
  usage: Please refer to license on github for more info.
</slate_header>
  */

package slate.common.encrypt

/**
 * Provides a simple facade for encryption using AES
 * @param _key : The secret key ( must be 16, or 32 bytes long )
 * @param _iv  : The iv ( must be 16 or 32 bytes long )
 */
class Encryptor(private val _key:String, private val _iv:String) {


  def encrypt(text:String ):String = {
    EncryptorAES.encrypt(_key, _iv, text)
  }


  def decrypt(encrypted:String):String = {
    EncryptorAES.decrypt(_key, _iv, encrypted)
  }
}

File: src/lib/kotlin/slatekit-common/src/main/kotlin/slatekit/common/encrypt/Encryptor.kt
/**
 * <slate_header>
 * url: www.slatekit.com
 * git: www.github.com/code-helix/slatekit
 * org: www.codehelix.co
 * author: Kishore Reddy
 * copyright: 2016 CodeHelix Solutions Inc.
 * license: refer to website and/or github
 * about: A tool-kit, utility library and server-backend
 * mantra: Simplicity above all else
 * </slate_header>
 */

package slatekit.common.encrypt

import slatekit.common.utils.B64

/**
 * Provides a simple facade for encryption using AES
 * @param key : The secret key ( must be 16, or 32 bytes long )
 * @param iv : The iv ( must be 16 or 32 bytes long )
 */
open class Encryptor(private val key: String,
                     private val iv: String,
                     private val b64: B64) {

    fun encrypt(text: String): String = EncryptorAES.encrypt(key, iv, text, b64)

    fun decrypt(encrypted: String): String = EncryptorAES.decrypt(key, iv, encrypted, b64)
}
